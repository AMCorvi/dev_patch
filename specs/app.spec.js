const db = require('../db/knex')
const request = require('supertest')

const app = require('../app')

describe('sticker api', () => {
  beforeAll(function(done) {
    db.migrate.latest().then(function() {
      db.seed.run()
    }).then(function () { done() })
  })

  test("404 response on invalid endpoint", function(done) {
    request(app)
      .get('/invalid')
      .expect("Content-Type", /json/)
      .expect(404)
      .then(res => {
        expect(res.body.message).toEqual('Not Found')
        expect(typeof res.body.error).toBe('object')
        done()
      })


  })

  test("findAll /api/v1/stickers", function (done) {

    request(app)
      .get('/api/v1/stickers')
      .set('Accept', 'application/json')
      .expect('Content-Type',/json/)
      .expect(200)
      .then(res => {
        expect(res.body).toBeInstanceOf(Object)
        expect(res.body.message).toMatchSnapshot()
        expect(Array.isArray(res.body.message)).toBe((true))
        done()
      })
  })

})
