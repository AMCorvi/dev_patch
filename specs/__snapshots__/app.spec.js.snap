// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sticker api findAll /api/v1/stickers 1`] = `Array []`;
